torchmetrics/functional/audio/pit.py

```diff
@@ -0,0 +1,181 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from importlib.util import find_spec
+from itertools import permutations
+from typing import Any, Callable, Dict, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+
+_ps_dict: dict = {}  # cache
+_ps_idx_dict: dict = {}  # cache
+
+_has_scipy = find_spec("scipy")
+
+
+def _find_best_perm_by_linear_sum_assignment(
+    metric_mtx: torch.Tensor,
+    eval_func: Union[torch.min, torch.max],
+) -> Tuple[Tensor, Tensor]:
+    from scipy.optimize import linear_sum_assignment
+    mmtx = metric_mtx.detach().cpu()
+    best_perm = torch.tensor([linear_sum_assignment(pwm, eval_func == torch.max)[1] for pwm in mmtx])
+    best_perm = best_perm.to(metric_mtx.device)
+    best_metric = torch.gather(metric_mtx, 2, best_perm[:, :, None]).mean([-1, -2])
+    return best_metric, best_perm  # shape [batch], shape [batch, spk]
+
+
+def _find_best_perm_by_exhuastive_method(metric_mtx: torch.Tensor,
```

Comment:
add docstring with descrition

Reply:
see https://github.com/PyTorchLightning/metrics/pull/384/commits/b452792349e84d5b4e3d1f2404fcc2bc5a35a64e

Reply:
also see https://github.com/PyTorchLightning/metrics/pull/384/commits/859e25e3218b8ca31b54a8f8a49bb95bb882117e

Reply:
The last one is the final one ...